### PR TITLE
fix(apigateway): type mismatch in C# when setting identitySources

### LINKED ISSUE #6538: RequestAuthorizer.IdentitySources not settable in .NET
If I try to set the property RequestAuthorizer.IdentitySources in my API Gateway setup, I can't create an instance of type IdentitySource (saying it needs to use the Authorization HTTP header). There is an empty constructor of the IdentitySource class and an IdentitySource.Header method which returns a string, but not an IdentitySource instance.

### Reproduction Steps

```
Authorizer = new RequestAuthorizer(this, "Authorizer", new RequestAuthorizerProps
                    {
                        IdentitySources = new[] {  ??????  }
                    })
```

### Environment
NuGet: Amazon.CDK.AWS.APIGateway 1.26.0

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by nija-at**
## Commit Message
fix(apigateway): type mismatch in C# when setting identitySources (#6649)

The `identitySources` property in `RequestAuthorizerProps` was
incorrectly set to the type `IdentitySource[]` instead of `string[]`.
This caused a type mismatch in languages with stricter type checks, such
as C#.

Added to `allowed-breaking-changes.txt` since this property would never
have worked in Java and C#, and is [not a breaking change in 
typescript][1].

[1]: https://github.com/aws/aws-cdk/blob/de1a36bb5d4c78e617f3b78a06b42dfbd45475dc/packages/%40aws-cdk/aws-apigateway/test/authorizers/integ.request-authorizer.ts#L26

fixes #6538
## End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nija-at** [comment]
cc @CaerusKaru

File: packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts [hunk 1]
```diff
@@ -190,7 +189,7 @@ export interface RequestAuthorizerProps extends LambdaAuthorizerProps {
    *
    * @see https://docs.aws.amazon.com/apigateway/api-reference/link-relation/authorizer-create/#identitySource
    */
-  readonly identitySources: IdentitySource[];
+  readonly identitySources: string[];
```
Comment (eladb) [inline packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts]: Why isn't this `IdentitySource`? It would make sense, no? What am I missing?

Comment (nija-at) [reply packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts]: The `IdentitySource` class is simply providing helper functions to build different types of identity source strings. 

The type here being `string[]` allows for a user to specify a identity source even if a corresponding helper method doesn't exist in `IdentitySource` (i.e., when APIGW adds more and we don't support yet).

It also stays consistent with `TokenAuthorizerProps` which uses `string` type for the `identitySource` property

Comment (eladb) [reply packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts]: Normally, we would keep this strong-typed and offer something like `IdentitySource.custom("bla")` or `IdentitySource.of("bla")` for future-proofing.

Comment (nija-at) [reply packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts]: Sure, but I would like to keep it consistent with the [`identitySource` property here](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-apigateway/lib/authorizers/lambda.ts#L135) which had already previously shipped and changing the type there would be breaking.

Would just adding a `IdentitySource.custom()` method satisfy? Do you have an different suggestion?

Comment (eladb) [reply packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts]: I am okay with changing this to a string.

**Pull request merged**